graphqlapi/src/test/java/io/stargate/graphql/schema/DmlSchemaCollectionsTest.java

```diff
@@ -0,0 +1,61 @@
+package io.stargate.graphql.schema;
+
+import static graphql.schema.GraphQLList.list;
+import static io.stargate.graphql.schema.SchemaAssertions.assertField;
+import static io.stargate.graphql.schema.SchemaAssertions.assertListFilter;
+import static io.stargate.graphql.schema.SchemaAssertions.assertMapFilter;
+
+import graphql.Scalars;
+import graphql.schema.GraphQLInputObjectType;
+import io.stargate.db.schema.Keyspace;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class DmlSchemaCollectionsTest extends DmlTestBase {
+
+  @Override
+  public Keyspace getKeyspace() {
+    return SampleKeyspaces.COLLECTIONS;
+  }
+
+  @Test
+  @DisplayName("Should surface list and set types")
+  public void listTest() {
+    GraphQLInputObjectType listFilterInput =
+        ((GraphQLInputObjectType) graphQlSchema.getType("ListIntFilterInput"));
+    assertListFilter(listFilterInput, Scalars.GraphQLInt);
+    assertField(graphQlSchema.getType("PkListTableFilterInput"), "l", listFilterInput);
+    assertField(graphQlSchema.getType("RegularListTableFilterInput"), "l", listFilterInput);
+    // CQL sets are translated to GraphQL lists, so these tables reuse the same type too:
+    assertField(graphQlSchema.getType("PkSetTableFilterInput"), "s", listFilterInput);
+    assertField(graphQlSchema.getType("RegularSetTableFilterInput"), "s", listFilterInput);
+  }
+
+  @Test
+  @DisplayName("Should surface map types")
+  public void mapTest() {
+    GraphQLInputObjectType mapFilterInput =
+        ((GraphQLInputObjectType) graphQlSchema.getType("ListEntryIntKeyStringValueFilterInput"));
```

Comment:
> ListEntry

Isn't this a map? Or am I reading this wrong?

Reply:
There is no map type in GraphQL, we represent them as lists of entries. For this particular example it looks like this:
```
// create table pk_map_table(m frozen<map<int,text>>)

type EntryIntKeyStringValue {
  key: Int!
  value: String
}

// Output type for results
type PkMapTable {
  m: [EntryIntKeyStringValue]
}

// Filter type for queries
input ListEntryIntKeyStringValueFilterInput {
  containsEntry: EntryIntKeyStringValueInput
  containsKey: Int
  containsValue: String
  eq: [EntryIntKeyStringValueInput]
  gt: [EntryIntKeyStringValueInput]
  gte: [EntryIntKeyStringValueInput]
  in: [[EntryIntKeyStringValueInput]]
  lt: [EntryIntKeyStringValueInput]
  lte: [EntryIntKeyStringValueInput]
  notEq: [EntryIntKeyStringValueInput]
}
```
The code that generates the filter name (`FieldFilterInputTypeCache.format()`) is not aware of those details, it just sees a "list of something". I think it would be possible to special case that to say "Map" instead of "List", but it seems low value, the name of the type should not be directly relevant to end users.